### PR TITLE
deleteQuestion end point code completed

### PR DESCRIPTION
**Pull request by upGradShashvat**
# Please enter the commit message for your changes. Lines starting
deleteQuestion end point code 
Please check

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java [hunk 1]
```diff
@@ -1,10 +1,15 @@
 package com.upgrad.quora.service.business;
 
+import com.upgrad.quora.service.common.ErrorMessage;
 import com.upgrad.quora.service.dao.QuestionDao;
+import com.upgrad.quora.service.entity.AnswerEntity;
```
Comment (shguptamts) [inline quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java:5]: remove this import, if we are not using AnswerEntity anywhere in QuestionService

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package com.upgrad.quora.service.dao;
 
+import com.upgrad.quora.service.entity.AnswerEntity;
```
Comment (shguptamts) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java]: remove this line, if answer entity is not used

File: quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java [hunk 3]
```diff
@@ -19,4 +24,29 @@ public QuestionEntity getQuestionByUuid(String questionUuid) throws InvalidQuest
...
+
+    /** Authorize the delete operation on a question
+     * Only an owner or admin can delete the question
+     * @param questionEntity  question entity, containing owner details
+     * @param userEntity    user performing delete operation
+     * @param questionUuid    question's uuid, to be deleted
+     * @return  true if user can delete the answer
+     * @throws AuthorizationFailedException exception is thrown if user is not allowed to delete the answer
+     */
+    public boolean authorize(QuestionEntity questionEntity, UserEntity userEntity, String questionUuid) throws AuthorizationFailedException {
```
Comment (shguptamts) [inline quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java]: remove questionUuid from the parameter as we are not using it .. I'll also remove same from AnswerService

**Review by shguptamts — CHANGES_REQUESTED** [review]
@upGradShashvat  
Kindly make specified 3 changes

File: quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java (hunk 3 above)
Comment (upGradShashvat) [reply quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java]: Made changes as per code review

File: quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java (hunk 2 above)
Comment (upGradShashvat) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java]: Made changes as per code review

File: quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java (hunk 1 above)
Comment (upGradShashvat) [reply quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java:5]: Made changes as per code review

**Review by shguptamts — APPROVED** [review]
looks good

**Comment by shguptamts** [comment]
add delete question functionality

**Pull request merged**